### PR TITLE
Fix read-cfpa command to read actual CFPA, not scratch

### PR DESCRIPTION
**Pull request by cbiffle**

### CODE REVIEW

File: lpc55_isp/src/bin/lpc55_flash.rs [hunk 1]
```diff
@@ -341,19 +350,46 @@ fn main() -> Result<()> {
...
+                    .context("reading CFPA ping page")?;
+                let pong = do_isp_read_memory(&mut *port, 0x9e000, 512)
+                    .context("reading CFPA pong page")?;
+                let ping_d = lpc55_areas::CFPAPage::from_bytes(ping[..].try_into().unwrap())?;
+                let pong_d = lpc55_areas::CFPAPage::from_bytes(pong[..].try_into().unwrap())?;
+                println!(
+                    "CFPA versions: ping={}, pong={}",
+                    ping_d.version, pong_d.version
+                );
+                if ping_d.version > pong_d.version {
```
Comment (kc8apf) [inline lpc55_isp/src/bin/lpc55_flash.rs:376]: Normally either ping or pong should be greater.  I have no idea what the ROM will do if they are equal.  Regardless, telling the user that the versions are equal seems important.

**Review by kc8apf — APPROVED** [review]

File: lpc55_isp/src/bin/lpc55_flash.rs [hunk 2]
```diff
@@ -341,19 +350,46 @@ fn main() -> Result<()> {
...
+                    CfpaChoice::Scratch => 0x9de00,
+                    CfpaChoice::Ping => 0x9e000,
+                    CfpaChoice::Pong => 0x9e200,
+                };
+                do_isp_read_memory(&mut *port, addr, 512)?
+            } else {
+                // Read ping and pong pages and only write out the latest one.
+                let ping = do_isp_read_memory(&mut *port, 0x9e000, 512)
+                    .context("reading CFPA ping page")?;
+                let pong = do_isp_read_memory(&mut *port, 0x9e000, 512)
```
Comment (inickles) [inline lpc55_isp/src/bin/lpc55_flash.rs:368]: Should the address here be 0x9e200? Looks to me like it's reading ping again.

Comment (kc8apf) [reply lpc55_isp/src/bin/lpc55_flash.rs:368]: Ayup.  Probably should use constants for those base addresses.

Comment (cbiffle) [reply lpc55_isp/src/bin/lpc55_flash.rs:368]: Thanks! Good catch.

File: lpc55_isp/src/bin/lpc55_flash.rs (hunk 1 above)
Comment (kc8apf) [reply lpc55_isp/src/bin/lpc55_flash.rs:376]: What happens if you have the same version but different content?  That's not worth holding this up for though.

**Pull request merged**